src/storage/heap_file.c

```diff
@@ -7177,6 +7184,10 @@ heap_scancache_quick_end (THREAD_ENTRY * thread_p, HEAP_SCANCACHE * scan_cache)
 	    {
 	      pgbuf_ordered_unfix (thread_p, &scan_cache->page_watcher);
 	    }
+	  if (scan_cache->old_page_watcher.pgptr != NULL)
+	    {
+	      pgbuf_ordered_unfix (thread_p, &scan_cache->old_page_watcher);
+	    }
```

Comment:
it won't happen. It's always unfixed on heap_next_internal().
```suggestion
	  assert (scan_cache->old_page_watcher.pgptr == NULL);
```